V2/src/fc_state_machine.cpp

```diff
@@ -2,57 +2,164 @@
 
 #include <fc_state_machine.h>
 #include <error_checker.h>
-
 #include <pin_defs.h>
+#include <analogs.h>
+
+#define CAP_THRESHOLD 20 // Move to another file, if they aren't already.
+
+void update_leds(void);
 
 #ifdef ENABLE_FC_TEST
 uint32_t state = FC_TEST;
 #else
-uint32_t state = FC_INIT;
+uint32_t state = FC_STANDBY;
 #endif
 
-uint32_t get_fc_state() {
-    // read-only access to state variable
+DigitalIn start(START); 
+
+// Valves.
+DigitalOut supply_v(SUPPLY_V);
+DigitalOut purge_v(PURGE_V);
+DigitalOut other1_v(VALVE3);
+DigitalOut other2_v(VALVE4);
+
+// Relays.
+DigitalOut start_r(START_R);
+DigitalOut motor_r(MOTOR_R);
+DigitalOut charge_r(CHARGE_R);
+DigitalOut cap_r(CAP_R);
+DigitalOut fcc_r(FCC_R);
+
+// LEDs.
+DigitalOut alarm_led(ALARM_LED);
+DigitalOut debug_led(DEBUG_LED);
+DigitalOut shut_led(SHUT_LED);
+DigitalOut run_led(RUN_LED);
+DigitalOut start_led(START_LED);
+
+uint32_t get_fc_state() // Read-only access to state variable.
+{
     return state;
 }
 
-// todo - other digital pin defs
-DigitalIn start(START);
+void fc_state_machine_thread() 
+{
+    fcc_r.write(true); // Solid state relay.
 
-// todo - define which timers we need
-Timer timer;
-
-void fc_state_machine_thread() {
-    while (true) {
-        if (check_all_errors()) {
-            state = FC_FAULT;
-        }
+    while (true) 
+    {
+        if (check_all_errors()) 
+            state = FC_ALARM;
 
-        if (state == FC_TEST) {
+        if (FC_TEST == state) 
```

Comment:
The test state needs to be included in the if/else group below, because otherwise it just switches to state = FC_ALARM when you hit that else statement below. I changed this in my branch while I was testing the fan control, but I don't know if merging is easier if it is changed here?

Reply:
Good catch. I'll add that to my next commit.